src/machine/usb/cdc/usbcdc.go

```diff
@@ -59,17 +59,27 @@ func (usbcdc *USBCDC) Buffered() int {
 	return int(usbcdc.rxBuffer.Used())
 }
 
+func (usbcdc *USBCDC) SetRXCallback(cb func(byte) bool) {
+	usbcdc.rxCallback = cb
+}
+
 // Receive handles adding data to the UART's data buffer.
 // Usually called by the IRQ handler for a machine.
 func (usbcdc *USBCDC) Receive(data byte) {
+	if usbcdc.rxCallback != nil {
+		if usbcdc.rxCallback(data) {
+			return
+		}
+	}
 	usbcdc.rxBuffer.Put(data)
```

Comment:
The callback receives data as an argument, and data is also added to the ringbuffer. This could potentially lead to some confusion in the handling of data. Do you have any thoughts on this?

Reply:
The idea is that the callback can return true if it successfully processed the byte and it doesn't need to be buffered. If it returns false, the byte would be buffered as if the callback did not exist.

Reply:
Understood.